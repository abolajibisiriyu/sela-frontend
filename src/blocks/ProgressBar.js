import styled from "styled-components";

const ProgressBar = styled.div`
  width: 100%;
  height: ${props => props.height || "28px"};
  background-color: ${props => props.bgColor || "#f2f2f2"};
  border-radius: 5px;
  position: relative;
  & > .progress-text {
    display: flex;
    align-items: center;
    justify-content: center;
    font-weight: bold;
    font-size: 0.6rem;
    position: absolute;
    width: 100%;
    height: 100%;
    font-size: 0.6rem;
    color: ${props =>
      props.progress && parseInt(props.progress, 10) >= 80 ? "#fff" : "#333"};
  }
  & > .innerBar {
    height: 100%;
    border-radius: ${props =>
      props.progress && parseInt(props.progress, 10) === 100
        ? "5px"
        : "5px 0px 0px 5px"};
    width: ${props => `${props.progress}%`};
    background-color: ${props => props.progressColor || "#2d9cdb"};
  }
`;

export default ProgressBar;
