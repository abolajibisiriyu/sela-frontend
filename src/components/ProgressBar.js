import React from "react";

import Progress from "../blocks/ProgressBar";

function ProgressBar({ height, progress }) {
  return (
    <Progress progress={progress}>
      <div className="progress-text">{progress}% progress</div>
      <div className="innerBar" />
    </Progress>
  );
}

export default ProgressBar;
