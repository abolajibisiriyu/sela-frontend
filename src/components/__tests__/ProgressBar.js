import React from "react";
import renderer from "react-test-renderer";

import ProgressBar from "../ProgressBar";

describe("ProgressBar Component", () => {
  it("renders correctly", () => {
    const component = renderer.create(<ProgressBar />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
