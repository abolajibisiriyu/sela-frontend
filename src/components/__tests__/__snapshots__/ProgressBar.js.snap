// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar Component renders correctly 1`] = `
<div
  className="sc-bdVaJa hlmBpM"
>
  <div
    className="progress-text"
  >
    % progress
  </div>
  <div
    className="innerBar"
  />
</div>
`;
